vipe/oozie/reader_handlers.py

```diff
@@ -0,0 +1,132 @@
+# limitations under the License.
+
+__author__ = "Mateusz Kobos mkobos@icm.edu.pl"
+
+from vipe.oozie.parse_utils import properties_to_dict
+from vipe.oozie.graph import DecisionCase, Decision, End, Kill, Fork, Join, \
+    Start, OtherAction, SubworkflowAction, JavaAction, StreamingMapReduceAction, \
+    JavaMapReduceAction, PigAction, HiveAction, FSAction, DistCPAction
+
+def handle_action(elem):
+    """
+    Returns:
+        a (name, Action) pair where name is a string.
+    """
+    name = elem.get('name')
+    type_ = __get_action_type(elem)
+    ok_node = elem.find('ok').get('to')
+    error_node = elem.find('error').get('to')
+    action_elem = elem.find(type_)
+    configuration = __get_configuration(action_elem)
+    if type_ == 'sub-workflow':
+        app_path = action_elem.find('app-path').text
+        return (name, SubworkflowAction(ok_node, error_node, configuration, 
+                                 app_path))
+    elif type_ == 'java':
+        main_class = action_elem.find('main-class').text
+        args = [e.text for e in action_elem.findall('arg')]
+        captures_output = action_elem.find('capture-output') is not None
+        return (name, JavaAction(ok_node, error_node, configuration, 
+                                 main_class, args, captures_output))
+    elif type_ == 'map-reduce':
+        streaming_elem = action_elem.find('streaming')
+        pipes_elem = action_elem.find('pipes')
+        if streaming_elem is not None:
+            mapper = streaming_elem.find('mapper')
+            reducer = streaming_elem.find('reducer')
+            return (name, StreamingMapReduceAction(ok_node, error_node, 
+                            configuration, mapper, reducer))
+        elif pipes_elem is not None:
+            ## pipes nodes are not parsed in detail since we haven't used them
+            ## yet
+            return (name, OtherAction(ok_node, error_node, configuration, 
+                                      action_elem.tag))
+        else:
+            return (name, JavaMapReduceAction(ok_node, error_node, 
+                                              configuration))
+    elif type_ == 'pig':
+        script = action_elem.find('script')
+        params = [e.text for e in action_elem.findall('param')]
+        args = [e.text for e in action_elem.findall('argument')]
+        return (name, PigAction(ok_node, error_node, configuration, 
+                                 script, params, args))
+    elif type_ == 'hive':
+        script = action_elem.find('script')
+        params = [e.text for e in action_elem.findall('param')]
+        return (name, HiveAction(ok_node, error_node, configuration, 
+                             script, params, args))
+    elif type_ == 'fs':
+        return (name, FSAction(ok_node, error_node))
+    elif type_ == 'distcp':
+        args = [e.text for e in action_elem.findall('arg')]
+        assert len(args) == 2
```

Comment:
This will not always be true, additional parameter such as `-pb` might be provided. Take a look at this example:

https://github.com/openaire/iis/tree/cdh5/iis-workflows/iis-workflows-top/src/main/resources/eu/dnetlib/iis/workflows/top/primary/main/oozie_app/workflow.xml

and `distcp_hbase_dump` element.

Maybe we should treat all `args` as generic and show all of them.